### ISSUE TITLE
[Bug] User data stored in Program Files on Windows

### ISSUE THREAD

**Issue by Davis-Klavins** [GeoDaCenter/geoda#2349]
On Windows, user data is stored in the same folder where software is installed (by default C:\Program Files\GeoDa Software\) and default permissions on install are set to full control for everyone which creates a security hole in case software restriction policies are used (permissions must be manually altered, exemptions in software restriction policies created which is time consuming). User data (everything writable by user) must be stored in user's AppData folder.

**Comment by lixun910** [comment]
Thanks @Davis-Klavins I was able to create a new installer using InnoSetup. Could you help to test if this new GeoDa installer for Windows 7+ works properly by installing it in users' AppData/local directory instead of C:\Program Files\GeoDa Software? You can download it from my personal dropbox (the link will be expired after 24 hours):

x64: https://www.dropbox.com/s/pd275akf21k4aw3/GeoDa_1.20_win7%2Bx64_Setup.zip?dl=0
x86: https://www.dropbox.com/s/8t4so9aawdkcvou/GeoDa_1.20_win7%2Bx86_Setup.zip?dl=0

Once it's tested and verified, we will publish the installers on our website. Thanks!

**Comment by Davis-Klavins** [comment]
Tested on Windows 10 x64. During installation, changed install location to C:\Program Files\GeoDa where software is supposed to be installed (C:\Program Files (x86)\GeoDa would be for 32-bit), only unins000.exe and unins000.dat got placed there, everything else went to C:\Users\[UserName]\AppData\Local\GeoDa. Executables (.exe, .dll etc.) must be placed in Program Files, user data (settings, temp files and everything else that is supposed to be writable by user) in user's AppData (in most cases, AppData/Roaming is used). Because locations can be changed, it's better to refer to them by environment variables:
%programfiles% (default C:\Program Files)
%programfiles(x86)% (default C:\Program Files(x86))
%appdata% (default C:\Users\[UserName]\AppData\Roaming)
Also, permissions on C:\Users\[UserName]\AppData\Local\GeoDa shouldn't be altered because it's supposed to be accessible only by that particular user, administrators and system. Though, it is wise to let the user choose either to install only for that particular user (in that case installation would go to C:\Users\[UserName]\AppData) or all users (to C:\Program Files), see https://jrsoftware.org/ishelp/index.php?topic=admininstallmode

**Comment by lixun910** [comment]
Thanks for your feedback and testing, @Davis-Klavins ! Here is a new installer following your suggestions:
x64: https://www.dropbox.com/s/vzr4p42k9uy68c3/GeoDa-Windows7%2B-x64-installer.zip?dl=0

* I was able to separate the user data (installed in AppData/Roaming/GeoDa) and program data (installed to user-specified location e.g. C:\Program Files) on Windows. So, there is no security hole, and the users can always get their data in a domain. New users can also run the installed GeoDa program and create their own user data under their own AppData/Roaming directory.

* I think your suggestion of letting users choose either to install for "all users" or "me" is great. However, I am not familiar with the `admininstallmode` in innosetup. Let me know if you would like to contribute to this feature: 
the current innosetup scripit is
https://github.com/lixun910/geoda/blob/v1.20/BuildTools/windows/installer/64bit/GeoDa-win7%2B.iss

Thanks for checking! Let me know any comments.

**Comment by Davis-Klavins** [comment]
Yes, settings are now separated. Only "C:\Program Files\GeoDa Software\web_plugins\logger.txt" still gets created by user, this also should go to %appdata%.
Permissions on C:\Program Files\GeoDa Software still have everyone full control added during install, should be removed, seems to be this line:
`Name: "{app}";  Permissions: everyone-full; Check: InitializeSetup`

A bit more reading (see https://stackoverflow.com/a/34331760) gave a clue that adding
`PrivilegesRequiredOverridesAllowed=commandline dialog`
under `[Setup]` would allow to display dialog to select install mode before UAC prompt which is displayed now when installation is run as non-administrator. Though, C++ redistributable installation anyway requires admin rights, thus it can be assumed that in most cases admin rights still would be necessary (except C++ redistributable is already installed as in case of upgrading to a newer version).

**Comment by lixun910** [comment]
Thank you, @Davis-Klavins ! These suggestions are great! Here is a new build: https://www.dropbox.com/s/vzr4p42k9uy68c3/GeoDa-Windows7%2B-x64-installer.zip?dl=0

* The logger.txt is located to %appdata%. 
* The "everyone-fullcontrol" permissions on C:\Program Files\GeoDa software is removed. (only read and execute)
* The %appdata%/GeoDa permissions are set to "users-modified"

I gave it a try following the stackoverflow post, but as you mentioned, it doesn't work if the installation is not run as administrator(c++ redistributable, windows registry). However, I think with the current installer, the administrator ((e.g. in a lab environment) can install GeoDa for all users, which will have their own geoda data under their own %appdata% directory.

**Comment by Davis-Klavins** [comment]
Checked, seems OK now.

> However, I think with the current installer, the administrator ((e.g. in a lab environment) can install GeoDa for all users, which will have their own geoda data under their own %appdata% directory.

Yes, it works that way.
